.github/workflows/rust.yml

```diff
@@ -293,6 +271,39 @@ jobs:
         run: |
           cargo -Zgitoxide -Zgit nextest run --locked
 
+  # Checks benchmark code generation, and runs each benchmark once.
+  # Fails if code generation fails, benchmarks panic, or generated weights are not valid Rust.
+  check-runtime-benchmarks:
+    # We always run benchmarks on our Linux reference machine
+    runs-on: ${{ fromJson(github.repository_owner == 'autonomys' &&
+      '"runs-on=${{ github.run_id }}/runner=self-hosted-ubuntu-22.04-x86-64"' || '"ubuntu-22.04"') }}
+
+    # Don't use the full 6 hours if a benchmark hangs
+    timeout-minutes: 120
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Install Protoc
+        uses: arduino/setup-protoc@c65c819552d16ad3c9b72d9dfd5ba5237b9c906b # v3.0.0
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Configure cache
+        uses: actions/cache@1bd1e32a3bdc45362d1e726936510720a7c30a57 # v4.2.0
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.toml') }}
+          restore-keys: |
+            ${{ runner.os }}-cargo-
+
+      - name: runtime-benchmark.sh check
+        run: |
+          scripts/runtime-benchmark.sh check
```

Comment:
I dont fully understand why `check` argument is required here

Reply:
The `check` argument activates cheaper compilation and the minimum number of benchmark iterations:
https://github.com/autonomys/subspace/blob/0594eccda6b7664c68020534bc0d4f21d2d7b7b9/scripts/runtime-benchmark.sh#L16-L22

Running more iterations or a `production` build just takes more CI time for no benefit, because we don't use the benchmark results. If we ever automate generating benchmark PRs, then that CI can pass `full` (or nothing) to the script.